### DISCUSSION TITLE
Should we re-write the example for ExpressionBodySyntax

### DISCUSSION THREAD

**Discussion by jobot0** [detekt/detekt#7604] [Q&A] [answered]
Hello!
Thanks to every contributors for this amazing project! It's a core tool for Kotlin developers and it looks like a gigantic work.

Before coming with my own MR, I would talk about the second example in https://detekt.dev/docs/rules/style/#expressionbodysyntax 

```kotlin
fun stuff() {
    return
        moreStuff()
            .getStuff()
            .stuffStuff()
}
```

Sure it's compliant but why it wouldn't be better to use `=` here? Both left and right looks pretty straightforward and it doesn't add much meaning compare to 

```kotlin
fun stuff() = moreStuff()
  .getStuff()
  .stuffStuff()
```

We could leverage this example to point the fact that have a body with a return type is more expressive for side-effects management inside the body where here we have a fluent api syntax without any expressed side-effect.

**Comment by cortinico** [comment, marked as answer]
There is a test that covers almost exactly your question:
https://github.com/detekt/detekt/blob/296dc789e04fa1b8e987574b70a06494e9a41df0/detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExpressionBodySyntaxSpec.kt#L120-L140

The reason that code is considered "compliant", is because the `includeLineWrapping` config property is set to false by default. If you set it to true, it would report also the code in the example you mentioned.
